sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,273 @@
+    its not possible to describe how the pathway moves, i.e. its length or its
+    shortening velocity.
+
+    >>> from sympy.physics.mechanics import ReferenceFrame
+    >>> from sympy.physics.vector import dynamicsymbols
+    >>> N = ReferenceFrame('N')
+    >>> q = dynamicsymbols('q')
+    >>> insertion.set_pos(origin, q * N.x)
+    >>> insertion.pos_from(origin)
+    q(t)*N.x
+
+    A pathway's length can be accessed via its ``length`` attribute.
+
+    >>> linear_pathway.length
+    sqrt(q(t)**2)
+
+    Note how what appears to be an overly-complex expression is returned. This
+    is actually required as it ensures that a pathway's length is always
+    positive.
+
+    A pathway's shortening velocity can be accessed similarly via its
+    ``shortening_velocity`` attribute.
+
+    >>> linear_pathway.shortening_velocity
+    -q(t)*Derivative(q(t), t)/sqrt(q(t)**2)
+
+    Parameters
+    ==========
+
+    origin : Point
+        The ``Point`` from which the pathway originates.
+    insertion : Point
+        The ``Point`` into which the pathway inserts.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        origin: Point,
+        insertion: Point,
+    ) -> None:
+        """Initializer for ``LinearPathway``.
+
+        Parameters
+        ==========
+
+        origin : Point
+            The ``Point`` from which the pathway originates.
+        insertion : Point
+            The ``Point`` into which the pathway inserts.
+
+        """
+        super().__init__(
+            origin=origin,
+            insertion=insertion,
+        )
+
+    def _true_length(self) -> Expr:
+        """Exact analytical expression for the pathway's length."""
+        length = self.insertion.pos_from(self.origin).magnitude()
+        return length
+
+    def _true_shortening_velocity(self) -> Expr:
+        """Exact analytical expression for the pathway's shortening velocity."""
+        relative_position = self.insertion.pos_from(self.origin)
+        if not relative_position:
+            return S.Zero
+        t = dynamicsymbols._t  # type: ignore
+        # A reference frame is needed to differentiate ``relative_position`` to
+        # ``relative_velocity`` so choose the first ``ReferenceFrame`` that
+        # ``relative_position`` is defined using.
+        frame = relative_position.args[0][1]
+        relative_velocity = relative_position.diff(t, frame)
+        shortening_velocity = -relative_velocity.dot(relative_position.normalize())
+        return shortening_velocity
+
+    def _unit_vector_to_origin(self) -> Vector:
+        """Unit vector towards the pathway's origin from its insertion."""
+        return self.origin.pos_from(self.insertion) / self.length
+
+    def _unit_vector_to_insertion(self) -> Vector:
+        """Unit vector towards the pathway's insertion from its origin."""
+        return self.insertion.pos_from(self.origin) / self.length
+
+    def forces(self, force: Symbol) -> list[tuple[Point, Vector]]:
```

Comment:
Isn't this something for the `Actuator` class to handle. Besides that I would prefer the usage of `loads` in that case, as that also includes torques. Please also, don't use `forcelist` as reference, but rather `loads`. Another thing is that each method actually requires the loads to be supplied as a list, not just `KanesMethod`.

Reply:
I would also when putting this in `ActuatorBase` consider it naming `to_loads`

Reply:
I originally tried implementing this as a method of the `ForceActuator` class. However, I realised that it is the specifics of the pathway that determines the directions (and number) of the forces in this list. So I think it needs to be associated with the pathway.

Noted about the renaming, I will address this by:
- [ ] Referencing `loads` instead.
- [ ] Referencing all methods, not just `KanesMethod`.
- [ ] Rename to something related to loads.

Reply:
I don't think either of `to_loads` or `as_loads` is quite correct for this method are it requires the `force` being passed. Would you be happy with `compute_loads`?

Reply:
> it requires the `force` being passed

This is a magnitude, not a force. `compute_loads` would be better, though I'm still not entirely convinced of it being in `PathWay`. Though we can of course have a method `to_loads` in `ActuatorBase`, where `ActuatorForce` just calls the `PathWay`.

Reply:
The direction why I'd like `ActuatorBase.to_loads` is, because it would make it easy to integrate within for example `System`.

Reply:
What I think we likely want here is to make this method private, i.e. `_compute_loads`. And then `ActuatorBase` can implement `to_loads`, which in the case of `ForceActuator` will just have:

```python
class ForceActuator(ActuatorBase):
    ....
    def to_loads(self) -> list[tuple[Point, Vector]]:
        return self.pathway._compute_loads(self.force)
```

Reply:
That is more sensible. Though how does this reflect in an actuator that dependents on the shortening velocity?

Reply:
Addressed in 3e2beee47220bf00a7d7d17f07d5d0a90c4851a0.